### ISSUE TITLE
[boost-asio] Add package option to enable io_uring support

### ISSUE THREAD

**Issue by gracicot** [microsoft/vcpkg#22064]
****

**Is your feature request related to a problem? Please describe.**

The next boost asio (version 1.78) has an option for enabling io_uring. It must be compiled with `BOOST_ASIO_HAS_IO_URING` and `BOOST_ASIO_DISABLE_EPOLL`

**Proposed solution**
An installation option to enable uring

**Describe alternatives you've considered**
The two options could also be enabled/disabled to selectively enable uring, epoll, or both.

**Additional context**
Boost 1.78 is out: https://www.boost.org/doc/libs/1_78_0/doc/html/boost_asio/history.html

**Comment by yurybura** [comment]
Boost.Asio is header only library. You can add these defines in your application.

**Comment by gracicot** [comment]
@yurybura As far as I know changing options would change the interface requirement for boost asio, which would apply to all packages using it. Simply using a define because it's header only don't cut it. If you have a dependency that use boost asio and you define those keyword in your project, you'll have ODR violation. The package manager should take care of that.

**Comment by yurybura** [comment]
The Boost packages are compatible with built-in CMake targets:
```
    find_package(Boost REQUIRED [COMPONENTS <libs>...])
    target_link_libraries(main PRIVATE Boost::boost Boost::<lib1> Boost::<lib2> ...)
```
vcpkg doesn't provide any other exported CMake target.

Due to boost-asio doesn't have binary files I think you can simply build your own application like this:
```
    find_package(Boost REQUIRED)
    target_link_libraries(main PRIVATE Boost::boost)
    target_compile_definitions(main 
        PRIVATE
            BOOST_ASIO_HAS_IO_URING=1
            BOOST_ASIO_DISABLE_EPOLL=1
    )
```

If other vcpkg package will use io_uring during the build this package must define have similar CMake code in `portfile.cmake`.

**Comment by gracicot** [comment]
I don't think this issue should be closed simply because there was no comments. I think it would be valuable to have a port feature for boost asio to enable io_uring. It is a header only library yes, but I don't think it's really relevant for this issue. If consistency, avoiding ODR violations and one day enable C++20 module builds is something vcpkg users want, it should be port feature.

**Comment by gracicot** [comment]
I still think a port feature is the best way to expose such functionality
